Client/Frontend/Fakespot/FakespotViewModel.swift

```diff
@@ -2,17 +2,94 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
-import Foundation
+import UIKit
+import Common
+import Shared
 
-class FakespotViewModel {
-    enum ViewState<T> {
+class FakespotViewModel: ObservableObject {
+    enum ViewState {
         case loading
-        case loaded(T)
+        case loaded(ProductAnalysisData?)
         case error(Error)
+
+        var viewElements: [ViewElement] {
+            var elements: [ViewElement] = []
+
+            switch self {
+            case .loading:
+                elements = [.loadingView]
+
+            case .loaded:
+                elements = [.reliabilityCard,
+                    .adjustRatingCard,
+                    .highlightsCard,
+//                    .qualityDeterminationCard, // card not created yet (FXIOS-6963)
+                    .settingsCard]
+
+            case .error:
+                // add error card
+                elements = [.settingsCard] // [.qualityDeterminationCard, .settingsCard]  // card not created yet (FXIOS-6963)
+            }
+
+            return elements
+        }
+
+        var productData: ProductAnalysisData? {
+            switch self {
+            case .loading, .error: return nil
+            case .loaded(let data): return data
+            }
+        }
     }
 
-    @Published private(set) var state: ViewState<ProductAnalysisData?> = .loading
+    enum ViewElement {
+        case loadingView
+//        case onboarding // card not created yet (FXIOS-7270)
+        case reliabilityCard
+        case adjustRatingCard
+        case highlightsCard
+//        case qualityDeterminationCard // card not created yet (FXIOS-6963)
+        case settingsCard
+        case noAnalysisCard
+        case messageCard
+    }
+
+    private(set) var state: ViewState = .loading {
+        didSet {
+            stateChangeClosure?()
+        }
+    }
     let shoppingProduct: ShoppingProduct
+    var stateChangeClosure: (() -> Void)?
```

Comment:
Minor: We already know this to be a closure, a more suitable name could be `onStateChange:`